packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -235,15 +235,26 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
     public submitSignal(content: any) { return this.connectionManager.submitSignal(content); }
 
     public flush() {
-        if (this.messageBuffer.length === 0) {
+        const batch = this.messageBuffer;
+        if (batch.length === 0) {
             return;
         }
 
+        this.messageBuffer = [];
+
         // The prepareFlush event allows listeners to append metadata to the batch prior to submission.
-        this.emit("prepareSend", this.messageBuffer);
+        this.emit("prepareSend", batch);
 
-        this.connectionManager.sendMessages(this.messageBuffer);
-        this.messageBuffer = [];
+        if (batch.length === 1) {
+            assert(batch[0].metadata?.batch === undefined, "no batch markup on single message");
+        } else {
+            assert(batch[0].metadata?.batch === true, "no start batch markup");
+            assert(batch[batch.length - 1].metadata?.batch === false, "no end batch markup");
```

Comment:
Should the deltamanager care about the batch metadata?

Reply:
I like defensive programming :) Given that we have (for now) two distinct paths (legacy and new one), that's the right place to have this check, I feel. But I intend to move it to runtime once we remove legacy path from loader.
But I can move it in this PR to runtime as well - let me know